pxtblocks/fields/field_utils.ts

```diff
@@ -167,6 +144,63 @@ namespace pxtblockly {
         return getAllFieldsCore(ws, f => f instanceof FieldTileset);
     }
 
+    export function needsTilemapUpgrade(ws: Blockly.Workspace) {
+        const allTiles = ws.getVariablesOfType(pxt.sprite.BLOCKLY_TILESET_TYPE).map(model => pxt.sprite.legacy.blocklyVariableToTile(model.name));
+        return !!allTiles.length;
+    }
+
+    export function upgradeTilemapsInWorkspace(ws: Blockly.Workspace, proj: pxt.TilemapProject) {
+        const allTiles = ws.getVariablesOfType(pxt.sprite.BLOCKLY_TILESET_TYPE).map(model => pxt.sprite.legacy.blocklyVariableToTile(model.name));
+        if (!allTiles.length) return;
+
+        Blockly.Events.disable();
```

Comment:
bug: Didn't spend time debugging it so I can finish reviewing, but this test imported incorrectly: https://makecode.com/_Libfj1hJ0euY

notice the third tilemap has the shrubs, but in the linked build it came out with the shrubs replaced with flowers

Reply:
Good catch! Fixed